scripts/js/lib/api/generateToc.test.ts

```diff
@@ -314,3 +318,43 @@ test("generate a toc without modules and releaes notes", () => {
     title: "My Quantum Project",
   });
 });
+
+test("returns nested entries grouped by major version", () => {
```

Comment:
Can you actually please get rid of this test and combine its logic into `TOC with separate release note files`? Like update the arguments to `Pkg.mock` so that you end up testing this behavior of grouping by major versions.